MakeHist/mumu.C

```diff
@@ -0,0 +1,337 @@
+       Long64_t ientry = LoadTree(jentry);
+       if (ientry < 0) break;
+       nb = fChain->GetEntry(jentry);   nbytes += nb;
+       // if (Cut(ientry) < 0) continue;
+
+       run_hist->Fill(run);
+
+       if(lep_pt[0] > 53 &&lep_pt[1] > 53){ //lep pT > 53GeV
+           if(dil_dR > 0.1){ // dR(lep,lep) > 0.1
+               if(fabs(lep_dz[0])<0.5 && fabs(lep_dz[1])<0.5){
+
+                   //Opposite sign dilepton
+                   if(lep_id[0] * lep_id[1] < 0){
+
+                       //if(dil_chosen != 0) continue;
+                       //particle id
+                       int id1 = lep_id[0];
+                       int id2 = lep_id[1];
+
+                       id_hist->Fill(id1);
+                       id_hist->Fill(id2);
+
+                       const double muon_mass = 0.1056583715;
+
+                       //Save leptons information
+                       vector<TLorentzVector> leptons;
+                       vector<Lepton> leptons_;
+                       for(unsigned lep_N=0; lep_N<2; ++lep_N){
+                           TLorentzVector lepton;
+                           TLorentzVector lepton_;
+                           lepton.SetPtEtaPhiM(lep_pt[lep_N],lep_eta[lep_N],lep_phi[lep_N],muon_mass);
+                           lepton_.SetPtEtaPhiM(lep_pt[lep_N],lep_eta[lep_N],lep_phi[lep_N],muon_mass);
+                           leptons.push_back(lepton);
+                           Lepton dummy = {lepton_, lep_pfIso[lep_N]};
+                           leptons_.push_back(dummy);
+                       }  // example(1): float FirstLeptonMass = leptons[0].M();
+                       ///// example(2): float FirstLeptonIso = leptons_[0].Iso;
+
+                       float Dil_dPhi[1] = {-999};
+                       float Dil_dR[1] = {-999};
+                       float Dil_mass[1] = {-999};
+
+                       float MET[1] = {-999};
+
+                       int Njet = 0;
+                       int Nbjet = 0;
+
+                       vector<TLorentzVector> Jets;
+                       vector<TLorentzVector> bJets;
+
+                       for(int NJ=0; NJ<nJets; NJ++){
+                           if(jet_pt[NJ] > 30){ //jet pT > 20GeV
+                               if(fabs(jet_eta[NJ]) < 5.0){ // jet |eta| <5.0
+
+                                   //di-lepton information (delta-Phi, delta-R, dilepton mass)
+                                   Dil_dPhi[0] = dil_dPhi;
+                                   Dil_dR[0] = dil_dR;
+                                   Dil_mass[0] = dil_mass;
+
+                                   //Missing ET
+                                   MET[0] = met_pt;
+
+                                   //Save all jet information 
+
+                                   TLorentzVector Jet;
+                                   Jet.SetPtEtaPhiE(jet_pt[NJ],jet_eta[NJ],jet_phi[NJ],jet_E[NJ]);
+                                   Jets.push_back(Jet);
```

Comment:
It will probably be better for the future to distinguish b-tagged jets from non-b-tagged jets exclusively, as the pT requirements are different and the current Jet struture will require you to make sure you are not doublecounting a b-tagged jet as a generic jet as well.

Reply:
I plan to change this part while changing the jet pT cut to 20 GeV.